pymc/distributions/continuous.py

```diff
@@ -856,6 +856,15 @@ def logcdf(value, loc, sigma):
             msg="sigma > 0",
         )
 
+    def icdf(value, loc, sigma):
+        res = Normal.icdf((value + 1.0) / 2.0, loc, sigma)
+        res = check_icdf_value(res, value)
+        return check_icdf_parameters(
+            res,
+            sigma > 0,
+            msg="sigma > 0",
+        )
```

Comment:
When removing the `res = check_icdf_value(res, value)`, it fails the test for `q < 0` or `q > 1`  since `Normal.dist()` and `pymc.logprob.basic.icdf()` don't check for it automatically. 

For example, in:

```
from pymc.logprob.basic import icdf

def icdf(value, loc, sigma):
    res = icdf(Normal.dist(loc, sigma), (value + 1.0) / 2.0)
    #res = check_icdf_value(res, value)
    return res
```

I get the following test failure:

<img width="874" alt="image" src="https://github.com/pymc-devs/pymc/assets/22087841/46382ce8-e87e-4407-b269-95db30f94d7e">

With `res = check_icdf_value(res, value)` uncommented the test pass successfully. 

Should I keep it or am I missing anything?

Reply:
Oh because we are transforming the value that goes into the Normal icdf. Yeah let's keep that. The sigma shouldn't be needed still

Reply:
Sounds good!! Thanks